zaza/openstack/charm_tests/nova/tests.py

```diff
@@ -154,6 +154,31 @@ def test_930_check_virsh_default_network(self):
                 model_name=self.model_name)
             self.assertFalse(int(run['Code']) == 0)
 
+    def test_940_enable_disable_actions(self):
+        """Test disable/enable actions on nova-compute units."""
+        nova_units = zaza.model.get_units('nova-compute',
+                                          model_name=self.model_name)
+
+        # Check that nova-compute services are enabled before testing
+        for service in self.nova_client.services.list(binary='nova-compute'):
+            self.assertEqual(service.status, 'enabled')
+
+        # Run 'disable' action on units
+        zaza.model.run_action_on_units([unit.name for unit in nova_units],
+                                       'disable')
+
+        # Check action results via nova API
+        for service in self.nova_client.services.list(binary='nova-compute'):
+            self.assertEqual(service.status, 'disabled')
+
+        # Run 'enable' action on units
+        zaza.model.run_action_on_units([unit.name for unit in nova_units],
+                                       'enable')
+
+        # Check action results via nova API
+        for service in self.nova_client.services.list(binary='nova-compute'):
+            self.assertEqual(service.status, 'enabled')
+
```

Comment:
So if this is a test for a new feature, this will break on old versions of the charm, specifically the stable series.  As we don't (yet) cut stable branches of these tests, it's normal practice to add a new test class with this test (for a new feature), and add the test to the tests/tests.yaml for the master branch of the charm.  Then, the new version gets the new tests, and the stable version of the charm does.  This means that we will continue to be able to backport changes to the stable series.

Reply:
Good point, thanks. I forgot that this was the procedure for new func tests.

Reply:
done